From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Wed, 7 Jan 2026 14:10:13 +0100
Subject: [PATCH] Revert "drivers: remoteproc: stm32mp15: check Cortex-M
 isolation"

This reverts commit b9f1c0820783436d45646fa50a62702f85d8fd62.

The MCKPROT is a system configuration that protects MCUSS clocks.
This must not be managed in remoteproc, as some peripherals use PLL3
as the parent clock.
Some SCMI services must first be implemented to manage the MCUSS clocks
before allowing the enabling of MCKPROT.

https://onedigi.atlassian.net/browse/DEL-9920

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I78214e4c482c3947fa36c0bde7cd2fe2eee133d4
---
 core/drivers/remoteproc/stm32_remoteproc.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/core/drivers/remoteproc/stm32_remoteproc.c b/core/drivers/remoteproc/stm32_remoteproc.c
index c99a47fe3..914474035 100644
--- a/core/drivers/remoteproc/stm32_remoteproc.c
+++ b/core/drivers/remoteproc/stm32_remoteproc.c
@@ -8,9 +8,6 @@
 #include <drivers/firewall_device.h>
 #include <drivers/rstctrl.h>
 #include <drivers/stm32_remoteproc.h>
-#ifdef CFG_STM32MP15
-#include <drivers/stm32mp1_rcc.h>
-#endif
 #include <keep.h>
 #include <kernel/cache_helpers.h>
 #include <kernel/dt_driver.h>
@@ -904,19 +901,6 @@ static TEE_Result stm32_rproc_probe(const void *fdt, int node,
 	stm32_rproc_a35ss_cfg(rproc);
 #endif
 
-#ifdef CFG_STM32MP15
-	if (!rproc->cdata->ns_loading) {
-		if (!stm32_rcc_is_secure()) {
-			if (IS_ENABLED(CFG_INSECURE))
-				IMSG("WARNING: insecure rproc support regarding RCC hardening");
-			else
-				panic("RCC secure hardening issue");
-		} else {
-			stm32_rcc_set_mckprot(true);
-		}
-	}
-#endif
-
 	if (!rproc->cdata->ns_loading)
 		SLIST_INSERT_HEAD(&rproc_list, rproc, link);
 
